src/overcooked_ai_py/mdp/overcooked_mdp.py

```diff
@@ -1953,30 +1959,119 @@ def process_for_player(primary_agent_idx):
         return final_obs_for_players
 
     @property
-    def featurize_state_shape(self):
-        return np.array([62])
-
-    def featurize_state(self, overcooked_state, mlam, horizon=400):
+    def featurize_state_shape(self, num_pots=2):
+        warnings.warn(
+            "Using the `featurize_state_shape` property is deprecated. Please use `get_featurize_state_shape` method instead",
+            DeprecationWarning
+        )
+        num_pot_features = 10
+        base_features = 28
+        total_features = self.num_players * (num_pots * num_pot_features + base_features)
+        return (total_features,)
+
+    def get_featurize_state_shape(self, num_pots=2):
+        num_pot_features = 10
+        base_features = 28
+        total_features = self.num_players * (num_pots * num_pot_features + base_features)
+        return (total_features,)
+
+    def featurize_state(self, overcooked_state, mlam, num_pots=2, **kwargs):
         """
-        Encode state with some manually designed features.
-        NOTE: currently works for just two players.
+        Encode state with some manually designed features. Works for arbitrary number of players
```

Comment:
Maybe add a comment explaining what num_pots does?

Reply:
Added an in depth docstring explaining what each param does and what features exactly are encoded